src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -275,7 +275,10 @@ protected NodeContext GetNode(string msbuildLocation, string commandLineArgs, in
 
                 // Create the node process
                 Process msbuildProcess = LaunchNode(msbuildLocation, commandLineArgs);
-                _processesToIgnore.Add(GetProcessesToIgnoreKey(hostHandshake, msbuildProcess.Id));
+                if (_componentHost.BuildParameters.EnableNodeReuse)
```

Comment:
Is this important for the change?

Reply:
I couldn't prove it, but document says Hashset is not threadsafe.